apiserver/facades/client/spaces/remove.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package spaces
+
+import (
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v3"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/core/permission"
+)
+
+// RemoveSpace removes the given space.
+// This method returns an error if: there are bindings, constraints on that space or the name is alpha.
+func (api *API) RemoveSpace(entities params.Entities) (params.ErrorResults, error) {
+	isAdmin, err := api.auth.HasPermission(permission.AdminAccess, api.backing.ModelTag())
+	if err != nil && !errors.IsNotFound(err) {
+		return params.ErrorResults{}, errors.Trace(err)
+	}
+	if !isAdmin {
+		return params.ErrorResults{}, common.ServerError(common.ErrPerm)
+	}
+	if err := api.check.ChangeAllowed(); err != nil {
+		return params.ErrorResults{}, errors.Trace(err)
+	}
+	if err := api.checkSupportsSpaces(); err != nil {
+		return params.ErrorResults{}, common.ServerError(errors.Trace(err))
+	}
+	results := params.ErrorResults{
+		Results: make([]params.ErrorResult, len(entities.Entities)),
+	}
+	for i, entity := range entities.Entities {
+		tag, err := names.ParseSpaceTag(entity.Tag)
+		if err != nil {
+			results.Results[i].Error = common.ServerError(err)
+			continue
+		}
+		if tag.Id() == network.AlphaSpaceName {
+			newErr := errors.New("the alpha space cannot be removed")
+			results.Results[i].Error = common.ServerError(newErr)
+			continue
+		}
+	}
+	return results, errors.NotSupportedf("RemoveSpace not implemented")
```

Comment:
Not supported?

Reply:
I see there's no mutation here, so this is OK for now.

Reply:
Will update in next patch